lib/puppet/file_system/windows.rb

```diff
@@ -89,6 +99,25 @@ def read_preserve_line_endings(path)
 
   private
 
+  def expand_path_cached(path)
+    @memoized_expanded_paths ||= {}
```

Comment:
what is the lifecycle of this cache? Risk of leaking between environments? What happens when files are removed or changed form (file to dir and vice versa)?